src/sql/workbench/browser/ui/infoBox/infoBox.ts

```diff
@@ -98,14 +113,89 @@ export class InfoBox extends Disposable implements IThemable {
 		this.updateStyle();
 	}
 
+	public get links(): azdata.LinkArea[] {
+		return this._links;
+	}
+
+	public set links(v: azdata.LinkArea[]) {
+		this._links = v ?? [];
+	}
+
 	public get text(): string {
 		return this._text;
 	}
 
 	public set text(text: string) {
 		if (this._text !== text) {
 			this._text = text;
-			this._textElement.innerText = text;
+
+			DOM.clearNode(this._textElement);
+			this._linkListenersDisposableStore.clear();
+
+			for (let i = 0; i < this._links.length; i++) {
+				const placeholderIndex = text.indexOf(`{${i}}`);
+				if (placeholderIndex < 0) {
+					this._logService.warn(`Could not find placeholder text {${i}} in text ${text}`);
+					// Just continue on so we at least show the rest of the text if just one was missed or something
+					continue;
+				}
+
+				// First insert any text from the start of the current string fragment up to the placeholder
+				let curText = text.slice(0, placeholderIndex);
+				if (curText) {
+					const span = DOM.$('span');
+					span.innerText = text.slice(0, placeholderIndex);
+					this._textElement.appendChild(span);
+				}
+
+				// Now insert the link element
+				const link = this._links[i];
+				const linkElement = DOM.$('a', {
```

Comment:
Link element refreshes the page and throws an error if the url is empty or invalid.

Reply:
What do you mean "refreshes" the page?

And when is this error thrown? On construction or on click? Because you would be passing in your own click handler to the Link element - since I believe right now it just assumes all links are valid. So you'd need to make your own to handle the "empty" links.

Reply:
Using this issue to track it https://github.com/microsoft/azuredatastudio/issues/19007